For some reason, rust always uses cc as the linker eventhough
we give them correct, working linker as arguments to the
configure script.

diff -Naur rustc-1.56.1-src.bak/compiler/rustc_target/src/spec/aarch64_alpine_linux_musl.rs rustc-1.56.1-src/compiler/rustc_target/src/spec/aarch64_alpine_linux_musl.rs
--- rustc-1.56.1-src.bak/compiler/rustc_target/src/spec/targets/aarch64_alpine_linux_musl.rs	2021-11-02 14:04:50.738628254 +0100
+++ rustc-1.56.1-src/compiler/rustc_target/src/spec/targets/aarch64_alpine_linux_musl.rs	2021-11-02 14:08:04.550460281 +0100
@@ -4,6 +4,7 @@
     let mut base = super::aarch64_unknown_linux_musl::target();
 
     base.llvm_target = "aarch64-alpine-linux-musl".into();
+    base.options.linker = Some("aarch64-alpine-linux-musl-gcc".into());
     base.options.vendor = "alpine".into();
     base.options.crt_static_default = false;
     base.options.static_position_independent_executables = true;
diff -Naur rustc-1.56.1-src.bak/compiler/rustc_target/src/spec/armv6_alpine_linux_musleabihf.rs rustc-1.56.1-src/compiler/rustc_target/src/spec/armv6_alpine_linux_musleabihf.rs
--- rustc-1.56.1-src.bak/compiler/rustc_target/src/spec/targets/armv6_alpine_linux_musleabihf.rs	2021-11-02 14:04:50.738628254 +0100
+++ rustc-1.56.1-src/compiler/rustc_target/src/spec/targets/armv6_alpine_linux_musleabihf.rs	2021-11-02 14:08:36.230218880 +0100
@@ -4,6 +4,7 @@
     let mut base = super::arm_unknown_linux_musleabihf::target();
 
     base.llvm_target = "armv6-alpine-linux-musleabihf".into();
+    base.options.linker = Some("armv6-alpine-linux-musleabihf-gcc".into());
     base.options.vendor = "alpine".into();
     base.options.crt_static_default = false;
     base.options.static_position_independent_executables = true;
diff -Naur rustc-1.56.1-src.bak/compiler/rustc_target/src/spec/armv7_alpine_linux_musleabihf.rs rustc-1.56.1-src/compiler/rustc_target/src/spec/armv7_alpine_linux_musleabihf.rs
--- rustc-1.56.1-src.bak/compiler/rustc_target/src/spec/targets/armv7_alpine_linux_musleabihf.rs	2021-11-02 14:04:50.738628254 +0100
+++ rustc-1.56.1-src/compiler/rustc_target/src/spec/targets/armv7_alpine_linux_musleabihf.rs	2021-11-02 14:09:01.046697127 +0100
@@ -4,6 +4,7 @@
     let mut base = super::armv7_unknown_linux_musleabihf::target();
 
     base.llvm_target = "armv7-alpine-linux-musleabihf".into();
+    base.options.linker = Some("armv7-alpine-linux-musleabihf-gcc".into());
     base.options.vendor = "alpine".into();
     base.options.crt_static_default = false;
     base.options.static_position_independent_executables = true;
